remme/tp/basic.py

```diff
@@ -30,6 +30,7 @@
 EMIT_EVENT = 'emit_event'
 PB_CLASS = 'pb_class'
 PROCESSOR = 'processor'
+VALIDATOR = 'validator'
```

Comment:
Why no further use of that? Performing validation in TPs too at least will serve as a sanity check of those validators.